### ISSUE TITLE
✨ Feature Request: Dedicated module for managing policies

### ISSUE THREAD

**Issue by nwerker** [ansible-collections/community.hashi_vault#332]
##### SUMMARY
Although I am sure it could be archived with the vault_write module, maybe it would be beneficial to have a dedicated vault policies module. This, in tandem with the existing PKI certificate module, would be perfect to streamline the management of vault client certificates for client authentication against vault.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

Not yet defined

##### ADDITIONAL INFORMATION

Additional module for the management (creation, management, deletion) of ACL Policies within vault. Supporting things like idempotence and maybe even ansible check mode.

**Comment by briantist** [comment]
Hi @nwerker , this certainly sounds like it would be a great addition!

For other readers I'll link to my comment on your other feature request so as not to repeat it:
- https://github.com/ansible-collections/community.hashi_vault/issues/331#issuecomment-1320726969

**tl;dr:** I'd love to help a contributor add this kind of thing but I don't know if I'll have time to do it myself.

**Comment by nwerker** [comment]
@briantist completely understandable as mentioned in #331!

**Comment by mathijswesterhof** [comment]
I've been working with this module for a while now and willing to make a pr for this if it's still wanted. I've also been planning on building one for entities (with groups)

**Comment by briantist** [comment]
> I've been working with this module for a while now and willing to make a pr for this if it's still wanted. I've also been planning on building one for entities (with groups)

That would be great! Please see the contributing guide in the collection docs and let me know if you have any questions or need help getting started.

**Comment by mathijswesterhof** [comment]
I'm currently working on this again now that HVAC error handling https://github.com/ansible-collections/community.hashi_vault/pull/391 is waiting to be reviewed. I saw that HVAC has a dedicated function for fetching policies and converting it with a hcl-parser if that is available. however the function only fetches from the api route `sys/policy` where `/sys/policies/acl` is more versatile and the policies route could be adjusted to also allow policies from rgp and egp (if the client has a enterprise subscription)

I'll have a look at how to solve this efficiently as the hvac client only implements parsed read. currently I've got:
 - [x] List policies
 - [ ] Get policy
     - [x] select policy by type (acl[default] rgp egp)
     - [ ] get hcs-parsed output for any policy type
 - [ ] Write policy
    - [x] changed mode
    - [x] no read before write (same as kv2)    
 - [ ] Delete policy

**Comment by AndrewJackson2020** [comment]
+1 for this request. 

At a minimum there should be some docs about how to do this with the vault_write module.
